File: 4107-find-missing-elements/find-missing-elements.py
class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        ans = []
        temp = [-1] * 101
        sm = float('inf')
        lg = float('-inf')

        for n in nums:
            if n < sm:
                sm = n
            if n > lg:
                lg = n
            temp[n] = 1

        for i in range(sm, lg):
            if temp[sm] == -1:
                ans.append(sm)
            sm += 1

        return ans
        

File: 4107-find-missing-elements/find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        vector<int> ans;
        vector<int> temp(101, -1);
        int sm=INT_MAX, lg = INT_MIN;
        for(int n: nums){
            if(n<sm) sm = n;
            if(n>lg) lg = n;
            temp[n] = 1;
        }

        while(sm<lg){
            if(temp[sm] == -1) ans.push_back(sm);
            sm++;
        }

        return ans;
    }
};